tfhe/src/integer/gpu/ciphertext/compressed_ciphertext_list.rs

```diff
@@ -40,11 +52,278 @@ impl CudaCompressedCiphertextList {
+    ///
+    ///     let cuda_decompression_key =
+    ///         compressed_decompression_key.decompress_to_cuda(radix_cks.parameters(), &streams);
+    ///
+    ///     let compression_key = compressed_compression_key.decompress();
+    ///
+    ///         let ct1 = radix_cks.encrypt(3_u32);
+    ///         let ct2 = radix_cks.encrypt_signed(-2);
+    ///         let ct3 = radix_cks.encrypt_bool(true);
+    ///
+    ///         let compressed = CompressedCiphertextListBuilder::new()
+    ///             .push(ct1)
+    ///             .push(ct2)
+    ///             .push(ct3)
+    ///             .build(&compression_key);
+    ///
+    ///         let cuda_compressed = compressed.to_cuda_compressed_ciphertext_list(&streams);
+    ///
+    ///         let d_decompressed1 = CudaUnsignedRadixCiphertext {
+    ///             ciphertext: cuda_compressed.get(0, &cuda_decompression_key, &streams),
+    ///         };
+    ///         let decompressed1 = d_decompressed1.to_radix_ciphertext(&streams);
+    ///         let decrypted: u32 = radix_cks.decrypt(&decompressed1);
+    ///         assert_eq!(decrypted, 3_u32);
+    ///
+    ///         let d_decompressed2 = CudaSignedRadixCiphertext {
+    ///             ciphertext: cuda_compressed.get(1, &cuda_decompression_key, &streams),
+    ///         };
+    ///         let decompressed2 = d_decompressed2.to_signed_radix_ciphertext(&streams);
+    ///         let decrypted: i32 = radix_cks.decrypt_signed(&decompressed2);
+    ///         assert_eq!(decrypted, -2);
+    ///
+    ///         let d_decompressed3 = CudaBooleanBlock::from_cuda_radix_ciphertext(
+    ///             cuda_compressed.get(2, &cuda_decompression_key, &streams),
+    ///         );
+    ///         let decompressed3 = d_decompressed3.to_boolean_block(&streams);
+    ///         let decrypted = radix_cks.decrypt_bool(&decompressed3);
+    ///         assert!(decrypted);
+    /// ```
+    pub fn to_cuda_compressed_ciphertext_list(
+        &self,
+        streams: &CudaStreams,
+    ) -> CudaCompressedCiphertextList {
+        let lwe_per_glwe = self.packed_list.lwe_per_glwe;
+
+        let modulus_switched_glwe_ciphertext_list =
+            &self.packed_list.modulus_switched_glwe_ciphertext_list;
+
+        let first_ct = modulus_switched_glwe_ciphertext_list.first().unwrap();
+        let storage_log_modulus = first_ct.packed_integers.log_modulus;
+        let initial_len = first_ct.packed_integers.initial_len;
+        let bodies_count = first_ct.bodies_count.0;
+
+        let message_modulus = self.packed_list.message_modulus;
+        let carry_modulus = self.packed_list.carry_modulus;
+
+        let first_block_info = CudaBlockInfo {
+            degree: Degree::new(message_modulus.0 * carry_modulus.0 - 1),
+            message_modulus,
+            carry_modulus,
+            pbs_order: self.packed_list.pbs_order,
+            noise_level: NoiseLevel::NOMINAL,
+        };
```

Comment:
Why is that need exactly, I didnt see that information on the CPU list

Reply:
Done.